mlflow/sklearn/utils.py

```diff
@@ -408,6 +415,168 @@ def _log_warning_for_artifacts(func_name, func_call, err):
     _logger.warning(msg)
 
 
+def _log_pretraining_metadata(estimator, *args, **kwargs):  # pylint: disable=unused-argument
```

Comment:
Refactored from `_log_pretraining_metadata` to batch the `log_params` and `set_tags` API calls together when possible.